grid2op/tests/test_ma.py

```diff
@@ -0,0 +1,518 @@
+            assert len(ma_env._subgrids_cls[space][agent].line_ex_to_subid) == ma_env._subgrids_cls[space][agent].n_line, add_msg
+            assert len(ma_env._subgrids_cls[space][agent].line_or_to_subid) == ma_env._subgrids_cls[space][agent].n_line, add_msg
+        
+        
+    def check_objects_to_subid(self, ma_env, domain, space = 'action'):
+        
+        # Verifies if sub ids are correct   
+        for agent in domain.keys():
+            
+            # Check if sub ids are smaller than the number of substations
+            assert (ma_env._subgrids_cls[space][agent].load_to_subid < ma_env._subgrids_cls[space][agent].n_sub).all()
+            assert (ma_env._subgrids_cls[space][agent].line_or_to_subid < ma_env._subgrids_cls[space][agent].n_sub).all()
+            assert (ma_env._subgrids_cls[space][agent].line_ex_to_subid < ma_env._subgrids_cls[space][agent].n_sub).all()
+            assert (ma_env._subgrids_cls[space][agent].storage_to_subid < ma_env._subgrids_cls[space][agent].n_sub).all()
+            assert (ma_env._subgrids_cls[space][agent].gen_to_subid < ma_env._subgrids_cls[space][agent].n_sub).all()
+            assert (ma_env._subgrids_cls[space][agent].interco_to_subid < ma_env._subgrids_cls[space][agent].n_sub).all()
+            if ma_env._subgrids_cls[space][agent].n_shunt:
+                assert (ma_env._subgrids_cls[space][agent].shunt_to_subid < ma_env._subgrids_cls[space][agent].n_sub).all()
+                
+            # Check if objects are correctly connected
+            for subid in range(ma_env._subgrids_cls[space][agent].n_sub):
+                dict_connected_objects = ma_env._subgrids_cls[space][agent].get_obj_connect_to(substation_id=subid)
+                
+                assert (np.where(ma_env._subgrids_cls[space][agent].load_to_subid == subid)\
+                    ==\
+                    dict_connected_objects["loads_id"]).all()
+                
+                assert (np.where(ma_env._subgrids_cls[space][agent].gen_to_subid == subid)\
+                    ==\
+                    dict_connected_objects["generators_id"]).all()
+                
+                assert (np.where(ma_env._subgrids_cls[space][agent].line_or_to_subid == subid)\
+                    ==\
+                    dict_connected_objects["lines_or_id"]).all()
+                
+                assert (np.where(ma_env._subgrids_cls[space][agent].line_ex_to_subid == subid)\
+                    ==\
+                    dict_connected_objects["lines_ex_id"]).all()
+                
+                assert (np.where(ma_env._subgrids_cls[space][agent].storage_to_subid == subid)\
+                    ==\
+                    dict_connected_objects["storages_id"]).all()
+            
+    def check_orig_ids(self, ma_env, domain : dict, space = 'action'):
+        # It tests if the origin ids are correct and
+        # we have the same with masks.
```

Comment:
origin of what ?

Are you sure it's not "test the id in the original grid" ?

In grid2op, "origin id" means "line_or_id"... Try avoid using the same word for 2 different concepts.

Reply:
Changed it